### ISSUE TITLE
TypeError: unhashable type: 'typing.ParamSpecArgs'

### ISSUE THREAD

**Issue by phorward** [viur-framework/viur-toolkit#16]
Hello,

when I want to use viur-toolkit in the latest version 0.1.2 in a Python 3.12 pipenv, I'm getting this error during local development server startup:

```
[2024-08-26 21:39:59,852] modules/__init__.py:50 [ERROR] Unable to import 'kurs'
Traceback (most recent call last):
  File "~/myproject/deploy/modules/__init__.py", line 34, in _import_modules
    _import = __import__(_module, globals(), locals(), [_module], level=1)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/myproject/deploy/modules/kurs.py", line 2, in <module>
    from viur.toolkit import Importable
  File "~/viur-toolkit/src/viur/toolkit/__init__.py", line 12, in <module>
    from .property import *
  File "~/viur-toolkit/src/viur/toolkit/property.py", line 13, in <module>
    class CachedProperty(t.Generic[Value, Args]):
  File "~/viur-toolkit/src/viur/toolkit/property.py", line 22, in CachedProperty
    args: t.Optional[Args.args] = None,
          ~~~~~~~~~~^^^^^^^^^^^
  File "~/.pyenv/versions/3.12.1/lib/python3.12/typing.py", line 384, in inner
    return func(*args, **kwds)
           ^^^^^^^^^^^^^^^^^^^
  File "~/.pyenv/versions/3.12.1/lib/python3.12/typing.py", line 496, in __getitem__
    return self._getitem(self, parameters)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.pyenv/versions/3.12.1/lib/python3.12/typing.py", line 730, in Optional
    return Union[arg, type(None)]
           ~~~~~^^^^^^^^^^^^^^^^^
  File "~/.pyenv/versions/3.12.1/lib/python3.12/typing.py", line 384, in inner
    return func(*args, **kwds)
           ^^^^^^^^^^^^^^^^^^^
  File "~/.pyenv/versions/3.12.1/lib/python3.12/typing.py", line 496, in __getitem__
    return self._getitem(self, parameters)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.pyenv/versions/3.12.1/lib/python3.12/typing.py", line 710, in Union
    parameters = _remove_dups_flatten(parameters)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "~/.pyenv/versions/3.12.1/lib/python3.12/typing.py", line 344, in _remove_dups_flatten
    return tuple(_deduplicate(params))
                 ^^^^^^^^^^^^^^^^^^^^
  File "~/.pyenv/versions/3.12.1/lib/python3.12/typing.py", line 319, in _deduplicate
    all_params = set(params)
                 ^^^^^^^^^^^
TypeError: unhashable type: 'typing.ParamSpecArgs'
```

In this case, I've used an editable viur-toolkit, but it also fails with the released version on PyPI.

**Comment by sveneberth** [comment]
Hello @phorward, I can't reproduce this problem with python `v3.11.9` and `v3.12.4`. You're using an older version (`v3.12.1`). Can you update you python version and test it again?
It looks like this was a bug in python, which got patched in `v3.12.3`: https://github.com/python/cpython/issues/112281 / [CHANGELOG](https://docs.python.domainunion.de/3/whatsnew/changelog.html#python-3-12-4-final:~:text=gh%2D112281%3A%20Allow%20creating%20union%20of%20types%20for%20typing.Annotated%20with%20unhashable%20metadata).

**Comment by sveneberth** [comment]
An update of the Python interpreter to the latest patch level has helped to solve the problem in all cases known to me. Therefore, in my opinion, it is not a problem in the toolkit.
